lib/board.rb

```diff
@@ -0,0 +1,37 @@
+class Board
+    attr_accessor :cells
+    def initialize
+        reset!
+    end
+    def display
+      #binding.pry
+      puts " #{cells[0]} | #{cells[1]} | #{cells[2]} "
+      puts "-----------"
+      puts " #{cells[3]} | #{cells[4]} | #{cells[5]} "
+      puts "-----------"
+      puts " #{cells[6]} | #{cells[7]} | #{cells[8]} "
+    end
+    def position(user_input)
+      cells[user_input.to_i-1]
+    end
+    def taken?(user_input)
+      position(user_input) != " "
+    end
+    def valid_move?(user_input)
+      user_input_to_i = user_input.to_i
+      user_input <= "9" && user_input >= "1" && !taken?(user_input)
+    end
+    def update(user_input, player)
+      cells[user_input.to_i-1]=player.token if !taken?(user_input) && valid_move?(user_input)
+      display
+    end
+    def full?
+      !cells.include?(" ")
+    end
+    def turn_count
```

Comment:
Use the highest level and most expressive iterator possible to solve a problem. [`count`](http://ruby-doc.org/core-2.2.0/Array.html#method-i-count) on the cells is perfect and the rest of this method is not needed. Entire body could read:

```ruby
def turn_count
  cells.count { |cell| cell == "X" || cell == "O" }
end
```